chapter3.adoc

```diff
@@ -16,42 +16,20 @@ Will be addressed as follows.
 
 The Debug Module Security Extension imposes security constraints and introduces non-backward-compatible changes. The presence of the Debug Module Security Extension can be determined by polling the allsecured/anysecured bits in dmstatus.
 
-[caption="Register {counter:rimage}: ", reftext="Register {rimage}"]
-[title="The `allsecured` and `anysecured` fields in dmstatus"]
-[id=dmstatus]
-[wavedrom, ,svg]
-....
-{reg: [
-  {bits:   20, name: 'defined in Debug Module'},
-  {bits:   1, name: 'allsecured'},
-  {bits:   1, name: 'anysecured'},
-  {bits:   3, name: 'defined in Debug Module'},
-  {bits:   7, name: '0'},
-], config:{lanes: 3, hspace:1024}}
-....
-
-[options="header"]
-|===============================================================================================================================================
-| Field      | Description                                                                                                | Access | Reset
-| allsecured |  The field is 1 when all currently selected hart are secured and not running in debuggable privilege levels | R    | 0    
-| anysecured |  The field is 1 when any currently selected hart is secured and not running in debuggable privilege levels  | R    | 0    
-|===============================================================================================================================================
-
 === Halt
 
 The hart can only be halted when it is running in privilege levels specified in <<dbgpriv>>. Otherwise, the halt request will remain pending. Consequently, the response to the external debugger request may take longer than one second, and this constraint must be eliminated.
 
-When machine mode is not permitted (`mdbgen` set to 0) to engage in debugging, the halt-on-reset (resethaltreq) operation must fail and set cmderr to 6 (security fault error).
 
 === Reset
 
-The reset operations must be safeguarded against various attacks. The RISC-V Debug Specification defines the hartreset and ndmreset operations, which reset selected harts or the entire system (excluding the Debug Module), respectively. The hartreset operation must be prohibited when machine mode is not allowed to be debugged. The security fault error will be raised if the operation is issued when `mdbgen` is 0. 
+The reset operations must be safeguarded against various attacks. The RISC-V Debug Specification defines the hartreset and haltresetreq operations, which reset selected harts. The reset operations must be prohibited when machine mode is not allowed to be debugged. The security fault error will be raised if the operation is issued when mdbgen[i] is 0. The debugger could check the error by polling allsecfault/anysecfault bits in dmstatus as specified in <<regdmstatus>>. 
```

Comment:
What is `haltresetreq`? I can't find any mention of it in the debug spec. If `resethaltreq` is meant, this is incorrect, it does not reset the selected hart, it requests that the selected hart boot from reset straight into debug mode.

Reply:
Right, it is mistake. Will describe it in halt subsection